test/logflare_web/open_telemetry_sampler_test.exs

```diff
@@ -1,19 +1,268 @@
 defmodule LogflareWeb.OpenTelemetrySamplerTest do
-  @moduledoc false
   use LogflareWeb.ConnCase
-  import LogflareWeb.OpenTelemetrySampler, only: [should_sample: 7]
+
+  alias LogflareWeb.OpenTelemetrySampler
 
   setup do
     prev_ingest = Application.get_env(:logflare, :ingest_sample_ratio)
     prev_endpoint = Application.get_env(:logflare, :endpoint_sample_ratio)
+    prev_metadata = Application.get_env(:logflare, :metadata)
+
     Application.put_env(:logflare, :ingest_sample_ratio, 1.0)
     Application.put_env(:logflare, :endpoint_sample_ratio, 1.0)
+    Application.put_env(:logflare, :metadata, [])
 
     on_exit(fn ->
       Application.put_env(:logflare, :ingest_sample_ratio, prev_ingest)
       Application.put_env(:logflare, :endpoint_sample_ratio, prev_endpoint)
+      Application.put_env(:logflare, :metadata, prev_metadata)
     end)
+
+    :ok
+  end
+
+  doctest LogflareWeb.OpenTelemetrySampler, import: true
+
+  describe "setup/1" do
+    test "delegates to trace id ratio based sampler" do
+      opts = %{probability: 0.5}
+      assert config = %{probability: 0.5} = OpenTelemetrySampler.setup(opts)
+
+      expected_config = :otel_sampler_trace_id_ratio_based.setup(0.5)
+      assert config == expected_config
+    end
```

Comment:
this looks like testing implementation.

Reply:
I agree, I removed in last push. I had no idea how to directly test setup function, but I am not just using it as the setup process of the other tests and I think that is good enough 🙏